File: lab5/с++/c#/Ellipse.cs
using System;

namespace GeometryLib
{
    // Клас еліпс — спадкоємець Figure
    public class Ellipse : Figure
    {
        private double a; // піввісь a
        private double b; // піввісь b

        // Конструктор з параметрами
        public Ellipse(double a, double b)
        {
            this.a = a;
            this.b = b;
        }

        // Реалізація властивості площі (πab)
        public override double Area => Math.PI * a * b;

        // Наближена формула довжини еліпса (Раман)
        public override double Perimeter =>
            Math.PI * (3 * (a + b) - Math.Sqrt((3 * a + b) * (a + 3 * b)));
    }
}

File: lab5/с++/Ellipse.cpp
#include "Ellipse.h"
#include <cmath>

Ellipse::Ellipse(double a, double b) : a(a), b(b) {}

double Ellipse::getArea() const {
    return M_PI * a * b;  // площа еліпса
}

double Ellipse::getPerimeter() const {
    return M_PI * (3 * (a + b) - sqrt((3 * a + b) * (a + 3 * b))); // довжина еліпса (наближено)
}